### PR TITLE
Bug 1814709: Don't revert S3 Encrypt changes by user

### PR DESCRIPTION
**Pull request by coreydaley**
Set Encrypt:true if the S3 storage configuration is blank
Don't revert Encrypt:false if set by the user

### CODE REVIEW

File: test/e2e/aws_test.go [hunk 1]
```diff
@@ -679,6 +679,27 @@ func TestAWSChangeS3Encryption(t *testing.T) {
 			t.Errorf("unable to find environment variable: wanted %s", val.Name)
 		}
 	}
+
+	// If a user sets Encrypt: false, don't overwrite it
```
Comment (dmage) [inline test/e2e/aws_test.go]: That still changes the behaviour of the operator. In the current version a customer can provide only `keyID` or `cloudFront`, the operator will populate everything else and set `encrypt` to `true`.

I don't think we can fix this problem without breaking backward compatibility and without introducing a new API field.

Comment (ricardomaraschini) [reply test/e2e/aws_test.go:683]: An idea, can't we make the encrypt field a pointer to a bool? At least I have seen this approach being used to differentiate between "user has set it to false" from "user did not set it". My two cents.

Comment (dmage) [reply test/e2e/aws_test.go:683]: No, it's a bad idea to have different logic for `null` and `false`. We've already discussed it,  @deads2k suggested to deprecate the existing field and add `insecureDisableEncryptedStorage`.

Comment (ricardomaraschini) [reply test/e2e/aws_test.go:683]: Makes sense.

**Pull request closed without merging**